### PR TITLE
Update integer_na.rst with multiple improvements.

### PR DESCRIPTION
**Pull request by Aryasarkar008**
updating and improving multiple problems in the documentation.

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/integer_na.rst [hunk 1]
```diff
@@ -1,17 +1,12 @@
 .. currentmodule:: pandas
 
-{{ header }}
```
Comment (mroeschke) [inline doc/source/user_guide/integer_na.rst:3]: This should be kept here

File: doc/source/user_guide/integer_na.rst [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
-
 .. _integer_na:
 
 **************************
 Nullable integer data type
 **************************
 
-.. note::
```
Comment (mroeschke) [inline doc/source/user_guide/integer_na.rst:11]: This should be kept here

File: doc/source/user_guide/integer_na.rst [hunk 3]
```diff
@@ -55,19 +50,15 @@ NumPy array.
 You can also pass the list-like object to the :class:`Series` constructor
 with the dtype.
 
-.. warning::
```
Comment (mroeschke) [inline doc/source/user_guide/integer_na.rst:58]: This should be kept here

**Comment by Aryasarkar008** [comment]
I have made the necessary changes as suggested by you. Please do check.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**